server/chat-plugins/helptickets.ts

```diff
@@ -505,10 +511,19 @@ export class HelpTicket extends Rooms.RoomGame {
 		const date = Chat.toTimestamp(new Date()).split(' ')[0];
 		void FS(`logs/tickets/${date.slice(0, -3)}.jsonl`).append(JSON.stringify(entry) + '\n');
 	}
-	static async getTextLogs(userid: ID, date?: string) {
+
+	/**
+	 * @param search [search key, search value] (ie ['userid', 'mia']
+	 * returns tickets where the userid property === mia)
+	 * If the [value] is omitted (index 1), searches just for tickets with the given property.
+	 */
+	static async getTextLogs(search: [string, string] | [string], date?: string) {
 		const results = [];
 		if (await checkRipgrepAvailability()) {
-			const args = [`-e`, `userid":"${userid}`, '--no-filename'];
+			const args = [
+				`-e`, search[1] ? `${search[0]}":"${search[1]}` : `${search[0]}":`,
```

Comment:
This will break if you try to find logs where a key equals `''` or `0`.
```suggestion
				`-e`, search.length > 1 ? `${search[0]}":"${search[1]}` : `${search[0]}":`,
```